src/models/chart-model/index.js

```diff
@@ -92,33 +103,107 @@ export default function createChartModel({
     previousConstraints: undefined,
     updateWithSettings: undefined,
     constraintsHaveChanged: undefined,
+    progressive: false,
   };
 
-  function updatePartial() {
-    meta.updateWithSettings = false;
-    requestAnimationFrame(() => {
-      // TODO: cancel requests as well to optimize???
-      // const startTime = Date.now();
-      trendLinesService.update();
+  let dataPages = [];
+  let enableDataPagesExtract = true;
```

Comment:
This variable is a little bit confusing.. can it be solved in another way maybe, or otherwise some comment explaining it's purpose/how it works would be good.

Reply:
This is to avoid one thing: we don't want data pages is extracted during progressive rendering.

Reply:
However we can remove this if we make sure that we restore layoutService data pages after each chunk rendered.